test/e2e/e2e.go

```diff
@@ -43,41 +48,34 @@ const (
 	retryInterval        = 1 * time.Second  // polling interval timer
 	retryTimeout         = 40 * time.Second // polling timeout
 	rolloutTimeout       = 10 * time.Minute
-	agnhostImage         = "registry.k8s.io/e2e-test-images/agnhost:2.26"
-	agnhostImageNew      = "registry.k8s.io/e2e-test-images/agnhost:2.53"
 	iperf3Image          = "quay.io/sronanrh/iperf"
 	redirectIP           = "123.123.123.123"
 	redirectPort         = "13337"
-	exContainerName      = "tcp-continuous-client"
 	defaultPodInterface  = "eth0"
 	udnPodInterface      = "ovn-udn1"
 )
 
 type podCondition = func(pod *v1.Pod) (bool, error)
 
 // setupHostRedirectPod
-func setupHostRedirectPod(f *framework.Framework, node *v1.Node, exContainerName string, isIPv6 bool) error {
-	_, _ = createClusterExternalContainer(exContainerName, externalContainerImage, []string{"-itd", "--privileged", "--network", externalContainerNetwork}, []string{})
-	nodeV4, nodeV6 := getContainerAddressesForNetwork(node.Name, externalContainerNetwork)
+func setupHostRedirectPod(f *framework.Framework, externalContainer provider.ExternalContainer, nodeName, nodeIP string, isIPv6 bool) error {
 	mask := 32
 	ipCmd := []string{"ip"}
-	nodeIP := nodeV4
 	if isIPv6 {
 		mask = 128
 		ipCmd = []string{"ip", "-6"}
-		nodeIP = nodeV6
 	}
-	cmd := []string{"docker", "exec", exContainerName}
+	cmd := []string{}
 	cmd = append(cmd, ipCmd...)
 	cmd = append(cmd, "route", "add", fmt.Sprintf("%s/%d", redirectIP, mask), "via", nodeIP)
-	_, err := runCommand(cmd...)
+	_, err := provider.Get().ExecExternalContainerCommand(externalContainer, cmd) // cleanup not needed because container is'nt reused between test specs
```

Comment:
`provider.Get().ExecExternalContainerCommand(externalContainer, cmd)` seems like it ought to be just `externalContainer.Run(cmd)`. Also, not sure what the comment means. Also, typo: "isn't"

Reply:
ya, i was thinking that too - can we do it as a follow up ?